Sources/_CryptoExtras/RSA/RSA_boring.swift

```diff
@@ -315,6 +344,215 @@ extension BoringSSLRSAPublicKey {
+            guard hash.withUnsafeBytes({ hashBufferPtr in
+                CCryptoBoringSSL_RSA_padding_add_PKCS1_PSS_mgf1(
+                    rsaPublicKey,
+                    &encodedMessage,
+                    hashBufferPtr.baseAddress,
+                    hashDigestType.dispatchTable,
+                    hashDigestType.dispatchTable,
+                    saltLength
+                )
+            }) == 1 else {
+                switch ERR_GET_REASON(CCryptoBoringSSL_ERR_get_error()) {
+                case RSA_R_DATA_TOO_LARGE_FOR_KEY_SIZE:
+                    throw _RSA.BlindSigning.Error.messageTooLong
+                default:
+                    throw CryptoKitError.internalBoringSSLError()
+                }
+            }
+
+            // 3. m = bytes_to_int(encoded_msg)
+            CCryptoBoringSSL_BN_bin2bn(&encodedMessage, encodedMessage.count, m)
+
+            // 4. c = is_coprime(m, n)
+            let n = CCryptoBoringSSL_RSA_get0_n(rsaPublicKey)
+            CCryptoBoringSSL_BN_gcd(gcd, m, n, bnCtx)
+            let c = CCryptoBoringSSL_BN_is_one(gcd) == 1
+
+            // 5. If c is false, raise an "invalid input" error and stop
+            guard c else {
+                throw _RSA.BlindSigning.Error.invalidInput
+            }
+
+            // 6. r = random_integer_uniform(1, n)
+            // 7. inv = inverse_mod(r, n)
+            // 8. If inverse_mod fails, raise a "blinding error" error and stop
+            // NOTE: We retry here until we get an appropriate r, which is suggested.
+            repeat {
+                guard CCryptoBoringSSL_BN_rand_range_ex(r, 1, n) == 1 else {
+                    throw CryptoKitError.internalBoringSSLError()
+                }
+            } while (CCryptoBoringSSL_BN_mod_inverse(inv, r, n, bnCtx) == nil)
+
+            // 9. x = RSAVP1(pk, r)
+            let e = CCryptoBoringSSL_RSA_get0_e(rsaPublicKey)
+            let montCtx = CCryptoBoringSSL_BN_MONT_CTX_new_for_modulus(n, bnCtx)
+            defer { CCryptoBoringSSL_BN_MONT_CTX_free(montCtx) }
+            guard CCryptoBoringSSL_BN_mod_exp_mont(x, r, e, n, bnCtx, montCtx) == 1 else {
+                throw CryptoKitError.internalBoringSSLError()
+            }
+
+            // 10. z = (m * x) mod n
+            guard CCryptoBoringSSL_BN_mod_mul(z, m, x, n, bnCtx) == 1 else {
+                throw CryptoKitError.internalBoringSSLError()
+            }
+
+            // 11. blinded_msg = int_to_bytes(z, modulus_len)
+            var blindedMessageBytes = [UInt8](repeating: 0, count: outputSize)
+            guard CCryptoBoringSSL_BN_bn2bin_padded(&blindedMessageBytes, outputSize, z) == 1 else {
+                throw CryptoKitError.internalBoringSSLError()
+            }
+
+            // 12. output blinded_msg, inv
+            let blindedMessage = Data(blindedMessageBytes)
+            var invBytes = [UInt8](repeating: 0, count: outputSize)
```

Comment:
This is another good spot for an `unsafeTemporaryAllocation`, as that way we can avoid the temporary `Data` object.